src/Collection/Collection.service.ts

```
@@ -23,18 +32,210 @@ import {
+   * It will also receive a signed message and its signature, so it can check and store it in the SlotUsageCheque
+   */
+  public async publishTPCollection(
+    dbCollection: ThirdPartyCollectionAttributes,
+    dbItems: ItemAttributes[],
+    signedMessage: string,
+    signature: string
+  ): Promise<{
+    collection: CollectionAttributes
+    items: FullItem[]
+  }> {
+    // For DCL collections, once a published collection item changes, the PUSH CHANGES button appears
+    // That will fire a /collections/${collectionId}/curation which will create a new CollectionCuration
+    // Subsequent changes will not show the push changes button, as it's already under_review
+
+    // For TP items, creations always exist. PUSH CHANGES should appear if the item has an approved ItemCuration and has changes in the Catalyst
+    // That should fire /items/:id/curation for each item that changed
+
+    // What we should migrate over here is the creation of the virtual CollectionCuration, as there'll always be a publish before a PUSH CHANGES
+
+    if (dbItems.length === 0) {
+      throw new InvalidRequestError('Tried to publish no TP items')
+    }
+
+    try {
+      ethers.utils.verifyMessage(signedMessage, signature) // Throws if invalid
+    } catch (error) {
+      throw new InvalidRequestError(
+        'Tried to publish TP items with an invalid signed message or signature'
+      )
+    }
+
+    const collectionId = dbItems[0].collection_id
+    if (!collectionId) {
+      throw new InvalidRequestError(
+        'Cannot publish items without a collection id'
+      )
+    }
+
+    const allTheSameCollection = dbItems.every(
+      (item) => item.collection_id === collectionId
+    )
+    if (!allTheSameCollection) {
+      throw new InvalidRequestError(
+        'Cannot publish items that belong to different collections'
+      )
+    }
```

What if we just compare that `dbCollection.id === item.collection_id` for each item? We could throw a single error too.